scipy/stats/_morestats.py

```diff
@@ -1274,6 +1289,7 @@ def _all(x):
 
     optimfunc = methods[method]
 
+    x = np.asarray(x)
```

Comment:
is there a reason for introducing this here rather than in the new `else` branch below?

Reply:
No, I will move it to the else branch.